pinot-query-planner/src/test/java/org/apache/pinot/query/QueryEnvironmentTest.java

```diff
@@ -96,28 +85,19 @@ public void testQueryToStages()
   }
 
   @Test
-  public void testQueryToRel()
-      throws Exception {
-    PlannerContext plannerContext = new PlannerContext();
-    String query = "SELECT * FROM a JOIN b ON a.col1 = b.col2 WHERE a.col3 >= 0";
-    SqlNode parsed = _queryEnvironment.parse(query, plannerContext);
-    SqlNode validated = _queryEnvironment.validate(parsed);
-    RelRoot relRoot = _queryEnvironment.toRelation(validated, plannerContext);
-    RelNode optimized = _queryEnvironment.optimize(relRoot, plannerContext);
-
-    // Assert that relational plan can be written into a ALL-ATTRIBUTE digest.
-    StringWriter sw = new StringWriter();
-    PrintWriter pw = new PrintWriter(sw);
-    RelWriter planWriter = new RelXmlWriter(pw, SqlExplainLevel.ALL_ATTRIBUTES);
-    optimized.explain(planWriter);
-    Assert.assertNotNull(sw.toString());
+  public void testQueryProjectFilterPushdownForJoin() {
+    String query = "SELECT a.col1, a.ts, b.col2, b.col3 FROM a JOIN b ON a.col1 = b.col2 "
+        + "WHERE a.col3 >= 0 AND a.col2 IN  ('a', 'b') AND b.col3 < 0";
+    QueryPlan queryPlan = _queryEnvironment.planQuery(query);
```

Comment:
Can we actually verify the multi-stage tree and also the logical RelNode tree to make sure all incremental stages of planning are tested ?